File: server/src/main/java/de/ingrid/igeserver/configuration/PreKeycloakConfig.java
package de.ingrid.igeserver.configuration;

import org.keycloak.adapters.springboot.KeycloakSpringBootConfigResolver;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class PreKeycloakConfig {

    /**
     * Use properties in application.properties instead of keycloak.json
     */
    @Bean
    public KeycloakSpringBootConfigResolver keycloakConfigResolver() {
        return new KeycloakSpringBootConfigResolver();
    }

}

File: server/src/main/java/de/ingrid/igeserver/configuration/PreKeycloakConfig.kt
package de.ingrid.igeserver.configuration

import org.keycloak.adapters.springboot.KeycloakSpringBootConfigResolver
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class PreKeycloakConfig {
    /**
     * Use properties in application.properties instead of keycloak.json
     */
    @Bean
    open fun keycloakConfigResolver(): KeycloakSpringBootConfigResolver {
        return KeycloakSpringBootConfigResolver()
    }
}
